src/Microsoft.DotNet.Internal.Tools.SynchronizePackageProps/Program.cs

```diff
@@ -0,0 +1,397 @@
+                    continue;
+                }
+
+                if (!Regex.IsMatch(repo, "^https://github.com/([^/]*)/(.*?)/?$"))
+                {
+                    WriteWarning($"Repository {repo} for dependency {name} is not github, so will not be synchronized");
+                    continue;
+                }
+
+                if (remoteDependencies.TryGetValue(repo, out Dependency existingDependency))
+                {
+                    if (existingDependency.CommitHash != commitHash)
+                    {
+                        WriteError(
+                            $"Incoherent dependency for repository {repo} detected, dependencies {dependency.Name} and {name} do not have the same commit hash");
+                        errors |= Errors.IncoherentDependencies;
+                    }
+                }
+                else
+                {
+                    remoteDependencies.Add(repo, new Dependency(repo, commitHash, name));
+                }
+            }
+
+            return (remoteDependencies, errors);
+        }
+
+        private static async Task<(Dictionary<string, string> value, Errors error)> FetchConsolidatedRemotePackages(
+            Dictionary<string, Dependency> remoteDependencies)
+        {
+            Errors errors = 0;
+            var collapsedRemotePackages = new Dictionary<string, string>();
+            Dictionary<string, Dictionary<string, string>>
+                remotePackages = await FetchRemotePackageVersions(remoteDependencies);
+
+            foreach ((string repo, Dictionary<string, string> versions) in remotePackages)
+            {
+                foreach ((string package, string version) in versions)
+                {
+                    foreach ((string otherRepo, Dictionary<string, string> otherVersions) in remotePackages)
+                    {
+                        if (otherVersions.TryGetValue(package, out string otherVersion) && otherVersion != version)
+                        {
+                            WriteError(
+                                $"Remote package inconsistency for '{package}'. Version '{version}' in '{repo}' and '{otherVersion}' in '{otherRepo}', no consistency possible. update-dependencies to get remote versions in agreement");
+                            errors |= Errors.RemotePackagesInconsistent;
+                            continue;
+                        }
+
+                        collapsedRemotePackages[package] = version;
+                    }
+                }
+            }
+
+            return (collapsedRemotePackages, errors);
+        }
+
+        private static async Task<Dictionary<string, Dictionary<string, string>>> FetchRemotePackageVersions(
+            Dictionary<string, Dependency> remoteDependencies)
+        {
+            var assembly = Assembly.GetEntryAssembly();
+            var client = new GitHubClient(new ProductHeaderValue(
+                "Microsoft.DotNet.Internal.Tools.SynchronizePackageProps",
```

Comment:
NIT: `assembly.FullName`

Reply:
Well, "Name". Full name includes all the version crap, which isn't legal in a product header. :-)